service/controller/resource/masters/create_deallocate_legacy_instance.go

```diff
@@ -118,8 +119,8 @@ func (r *Resource) getRunningInstances(ctx context.Context, resourceGroup string
 	return instancesRunning, nil
 }
 
-func (r *Resource) getVMSS(ctx context.Context, resourceGroup string, vmssName string) (*compute.VirtualMachineScaleSet, error) {
-	c, err := r.GetScaleSetsClient(ctx)
+func (r *Resource) getVMSS(ctx context.Context, customObject providerv1alpha1.AzureConfig, resourceGroup string, vmssName string) (*compute.VirtualMachineScaleSet, error) {
```

Comment:
Not sure if it's better for these functions to receive the `AzureConfig` or the client so they don't have to create it. What do you think? It feels like passing the client we make explicit that they just need the client, not the `AzureConfig`.

Reply:
This part of the code is part of Flatcar migration, didn't really want to get into refactoring it, as it will get thrown out anyway.